Add plugin: JIRA links shortener

rplatonovs opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/rplatonovs/obsidian-jira-links-shortener

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[const jiraRegex = /^http.*\/browse\/([A-Z]+-\d+)/g;](https://github.com/rplatonovs/obsidian-jira-links-shortener/blob/97912e0a4c067aabe207a88e3a7aacd756c7e804/main.ts#L12)
This regex is a bit to generic, and could trigger with other URL's
It would be better if you let users configure their JIRA instance URL.

rplatonovs commented:
@joethei fair enough, I added the configuration you suggested. I also identified and fixed the case were query string can be lost from the URL. A few unit tests added for the core function that does the transformation of a pasted text.
